lib/Types.mli

```diff
@@ -0,0 +1,5 @@
+type rendering_mode = Color | TextMarkers
+type block_origin = Mine | Their | None
```

Comment:
I'm not a fan of this module.

These types make more sense closer to where they are used and do not appear general enough to be used in several contextes. `block_origin` for example appears only in `Block`, which is now harder to read with this type in an other module. I think this applies to every types in this module.

An other example is that `rendering_mode` do not make sense inside `Block`, which depends on this module for other things.

Reply:
Purpose: Avoid dependency cycles
But try something else.